### ISSUE TITLE
Feature request: Pre-signed URLs

### ISSUE THREAD

**Issue by fhickman** [joshdholtz/fastlane-plugin-s3#4]
Lots of our app distribution is for internal-only Enterprise-licensed builds.

For S3 buckets that we do not wish to make publicly readable, it would be great to generate the URLs within the .plist/.html using pre-signed URLs. You could supply a parameter for the expiration time of the URLs, or default it to 1 year.

(Sorry, if I knew a lick of Ruby I'd generate a PR myself. :) )

**Comment by joshdholtz** [comment]
@fhickman No worries! I will add this to list of features to implement 😊

**Comment by joshdholtz** [comment]
Closed because AWS S3 SDK only allows max of 1 week... See #14

**Comment by duro** [comment]
@fhickman and @joshdholtz I was able to start getting around this by putting CloudFront in-front of the S3 bucket and using a private ACL. I then used Lambda@Edge to implement some kind of authentication in front if hitting the bucket (simple http-auth for now, but it could be made more complex)

However, I am running into a new challenge, which is that the hostname is baked into the `apk_url` param in a way that I am finding trouble overriding at the template level.

Any suggestions on how I might do that?
